charts/opentelemetry-operator/values.yaml

```diff
@@ -354,6 +357,12 @@ securityContext:
   runAsUser: 65532
   fsGroup: 65532
 
+# -- Determines how the operator pod mounts the service account token.
+# When set to false, automounting of the service account token is disabled, and the token is manually mounted
+# using a volume and volumeMounts in the pod spec by the chart.
+# This is useful for security hardening, as it prevents the pod from automatically having access to the service account token unless explicitly mounted.
+automountServiceAccountToken: true
```

Comment:
do we need both of these?

Reply:
I wouldn't say we "need" two options for service account vs workload. I was going for more flexibility for any advanced use cases. But unifying both object's setting with one config for simplicity would also be okay.

Reply:
Yeah i think we should just have one to keep it easy IMO

Reply:
done!